task-library/AHV.py

```diff
@@ -0,0 +1,187 @@
+        except Exception:
+            print(resp)
+            raise RestUtilException("Rest API request failed : %s" % resp.reason)
+
+def _do_get(url, cookies=None, params=None, auth=None, timeout=120):
+    try:
+        session = RestUtil(BASE_URL).get_session()
+        headers = {'Content-Type': 'application/json'}
+        for i in range(3):
+            resp = session.get(url, params=params,
+                            auth=_auth(), timeout=timeout,
+                            headers=headers,cookies=cookies, verify=False)
+            if resp.status_code in [500,503] :
+                print("Sleeping for 60s because of status code",resp.status_code)
+                time.sleep(60)
+            else:
+                break
+        return _wrap(resp),resp
+    except Exception:
+        print("Rest API GET request failed")
+        raise
+
+def _do_post(url,params=None, cookies=None,auth=None, timeout=120):
+    try:
+        session = RestUtil(BASE_URL).get_session()
+        headers = {'Content-Type': 'application/json'}
+        for i in range(3):
+            resp = session.post(url, data=json.dumps(params),
+                            auth=_auth(), timeout=timeout,
+                            headers=headers, cookies=cookies, verify=False)
+            if resp.status_code in [500]:
+                time.sleep(300)
+            else:
+                break
+        #print(resp.json())
+        return _wrap(resp),resp
+    except Exception:
+        print("Rest API POST request failed")
+        raise
+
+def _do_put(url, params=None, auth=None, timeout=120):
+    try:
+        session = RestUtil(BASE_URL).get_session()
+        headers = {'Content-Type': 'application/json'}
+        for i in range(3):
+            resp = session.put(url, data=json.dumps(params),
+                            auth=_auth(), timeout=timeout,
+                            headers=headers, verify=False)
+            if resp.status_code in [500] :
+                time.sleep(300)
+            else:
+                break
+        return _wrap(resp)
+    except Exception:
+        print("Rest API PUT request failed")
+        raise
+
+def _do_delete(url, params=None, auth=None, timeout=120):
+    try:
+        session = RestUtil(BASE_URL).get_session()
+        headers = {'Content-Type': 'application/json'}
+        for i in range(3):
+            resp = session.delete(url, auth=_auth(), timeout=timeout,
+                              headers=headers, verify=False)
+            if resp.status_code in [500] :
+                time.sleep(300)
+            else:
+                break
+        return _wrap(resp)
+    except Exception:
+        print("Rest API DELETE request failed")
+        raise
+
+def _get_session(server):
+    http_req_adapter = requests.adapters.HTTPAdapter(max_retries=3, pool_maxsize=30, pool_connections=1)
+    s = requests.Session()
+    s.mount(server, http_req_adapter)
+    return s
+
+class RestUtil(object):
```

Comment:
Please add doc_string which will detailed about class's purpose, in args, return value etc..